pandas/_libs/lib.pyx

```diff
@@ -2354,8 +2387,7 @@ def maybe_convert_numeric(
             seen.float_ = True
         else:
             try:
-                floatify(val, &fval, &maybe_int)
-
+                floatify(val, &fval, &maybe_int, dsep[0], tsep[0])
```

Comment:
Is this reachable if the length of dsep or tsep is every zero? i.e. if someone did `thousands=""` seems like this could segfault?

Reply:
Yes you're right, I'll add a validation check to make sure both tsep and dsep will be at least length 1

Reply:
Changed the check on line 2227 to filter out zero length separators, the validation checks may need changing based on the decision re. multi width characters but the segfault should be fixed.